### PR TITLE
BE-976 Adding previous ids functionality

### PR DESCRIPTION
**Pull request by josecordaz**
**Description:**

Adding previous ids functionality.

### CODE REVIEW

File: cmd/upload.go [hunk 1]
```diff
@@ -48,4 +64,5 @@ func init() {
 	rootCmd.AddCommand(uploadCmd)
 	uploadCmd.Flags().Bool("progress", true, "show progress bar")
 	uploadCmd.Flags().Bool("debug", false, "turn on transfer debugging")
+	uploadCmd.Flags().Bool("dry-run", false, "no upload job.json")
```
Comment (pec1985) [inline cmd/upload.go]: I would change the wording to "does not upload the job.json to aws"

File: pkg/upload/upload.go [hunk 2]
```diff
@@ -608,17 +609,20 @@ func SendJob(ctx context.Context, logger log.Logger, dir string, debug bool, pro
...
+				StartedAt:      datetime.ISODateFromTime(started),
+				EndedAt:        datetime.ISODate(),
+				CustomerID:     customerID,
+				CustomerName:   customerName,
+				Metadata:       metadata,
+				UploadSize:     zipFileStats.Size(),
+				Agent:          sysinfo.GetSystemInfo(),
 			}))
 
+			config.PreviousJobIDs = append(config.PreviousJobIDs, jobid)
```
Comment (pec1985) [inline pkg/upload/upload.go]: This should be added at the end of a successful upload

File: pkg/upload/upload.go [hunk 3]
```diff
@@ -629,14 +633,27 @@ func SendJob(ctx context.Context, logger log.Logger, dir string, debug bool, pro
...
-				Bucket:  aws.String(fetcher.bucket),
-				Key:     aws.String(fetcher.path + "/job.json"),
-				Body:    bytes.NewReader(buffer),
-				Tagging: aws.String(tagging),
-			})
+			if dryrun {
+				var v map[string]interface{}
+				err := json.Unmarshal(buffer, &v)
+				if err != nil {
+					panic(err)
```
Comment (pec1985) [inline pkg/upload/upload.go]: Maybe `return err` instead of panic?

File: pkg/upload/upload.go [hunk 4]
```diff
@@ -629,14 +633,27 @@ func SendJob(ctx context.Context, logger log.Logger, dir string, debug bool, pro
...
+				}
+				res, err := json.MarshalIndent(v, "", " ")
+				if err != nil {
+					panic(err)
```
Comment (pec1985) [inline pkg/upload/upload.go]: Maybe `return err` instead of panic?

File: pkg/upload/upload.go [hunk 5]
```diff
@@ -629,14 +633,27 @@ func SendJob(ctx context.Context, logger log.Logger, dir string, debug bool, pro
...
+				}
+				fmt.Println(string(res))
```
Comment (pec1985) [inline pkg/upload/upload.go]: I think a more descriptive log statement would be more useful. Something like this
```
fmt.Println("job.json content:")
fmt.Println(string(res))
```
Or similar

File: pkg/upload/upload.go (hunk 3 above)
Comment (jhaynie) [reply pkg/upload/upload.go]: yes, please don't panic. panic should be reserved in cases where it's a programmer logic error and we can't do anything about it.

**Pull request merged**